action.yaml

```diff
@@ -57,71 +57,82 @@ runs:
   steps:
     - name: Enable problem-matcher
       shell: bash
+      env:
+        format: ${{ inputs.format }}
+        disable_matcher: ${{ inputs.disable_matcher }}
       run: |
-        problem_matcher_file="${{ github.action_path }}/.github/problem-matcher-${{ inputs.format }}.json"
-        if [[ ${{ inputs.disable_matcher }} != "true" && -f "$problem_matcher_file"  ]]; then
+        problem_matcher_file="${GITHUB_ACTION_PATH}/.github/problem-matcher-${format}.json"
```

Comment:
For `${{ github.action_path }}` and `${{ runner.os }}`. I wanted to avoid mixing github expressions and environment variables. In this context the environment variables were available so I opted to use them.

Also, it's a "better safe than sorry" type idea. While I believe that these can't be altered externally if they can one day, then they won't be a source of injection.

I'm open to reverting those back to the way they were if you'd like. Just let me know.

Reply:
I would prefer not to change more than what's needed.

If GitHub changes the behavior of these (and outputs (other comments)) everyone that uses actions will have problems.

Reply:
I have reverted back to using `${{ github.action_path }}` and `${{ runner.os }}`